### DISCUSSION TITLE
On add new item into "ObservableMap", functional component is not re rendering when the accessing item is dynamically created

### DISCUSSION THREAD

**Discussion by muneerashaik** [mobxjs/mobx#3497] [Q&A]
**Facing Issue**:

I am using an observable map to store the tasks. I have a use case where the task is added dynamically after accessing it.
I have an "arrow function" and "normal function" in my store to access the item. When I use "normal function", rendering is as expected otherwise there is no re-rendering happening

**How to reproduce the issue**:

The minimal example to reproduce the issue
https://codesandbox.io/s/flamboyant-lake-r34r1x?file=/src/App.tsx

**Steps to follow to reproduce**:

- There are two methods in TaskStore
  1. getTask
  2. getTaskArrowFunction
- if we use getTask getting issue

**Mobx versions**:

- "mobx": "5.15.0",
- "mobx-react": "6.1.4"

**Comment by urugator** [comment]
Action's are for mutating state, not for reading it as part of derivations (`observer`/`autorun`/`computed`/etc).
